### PR TITLE
Caching and splitting much better

### PR DESCRIPTION
**Pull request by j8seangel**
After a cool conversation on how @edbrett is doing code splitting in gfw, I thought it worth it to include in all of climate watch projects too.

This PR is going to include.
- Split every node_module dependency
- Proper code splitting without having a huge main.js
- Long-term cache for the static.js files

So now everything will work smoother! 

Once is merged I this that worths it to move to existing projects.

### CODE REVIEW

**Comment by Bluesmile82** [comment]
It's a cool idea but won't this affect the tree shaking?

**Comment by j8seangel** [comment]
Included also hot module reloading to avoid losing the state on every change refresh.

**Comment by j8seangel** [comment]
and it shouldn't affect the tree shaking.

File: config/environments/production.rb [hunk 1]
```diff
@@ -21,10 +25,10 @@
 
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
-  config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
+  # config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
```
Comment (edbrett) [inline config/environments/production.rb:28]: do you still want to disable these?

File: config/environments/production.rb [hunk 2]
```diff
@@ -13,6 +13,10 @@
   # Full error reports are disabled and caching is turned on.
   config.consider_all_requests_local       = false
   config.action_controller.perform_caching = true
+  config.serve_static_assets = true
+  config.public_file_server.headers = {
+    'Cache-Control' => "public, max-age=#{15.days.seconds.to_i}"
```
Comment (edbrett) [inline config/environments/production.rb:18]: only 15 days?

File: package.json [hunk 3]
```diff
@@ -24,7 +24,7 @@
     "react-css-themr": "2.1.2",
     "react-dom": "16.3.0",
     "react-redux": "5.0.5",
-    "react-universal-component": "3.0.0",
+    "react-universal-component": "5.0.0-alpha.0",
```
Comment (edbrett) [inline package.json:27]: Good idea.

File: config/environments/production.rb (hunk 1 above)
Comment (j8seangel) [reply config/environments/production.rb:28]: It was needed to serve the assets in production

File: config/environments/production.rb (hunk 2 above)
Comment (j8seangel) [reply config/environments/production.rb:18]: what is best ?

**Pull request closed without merging**